### PR TITLE
provision,builder: Add kubernetes platform-builder

### PR DESCRIPTION
**Pull request by guilhermebr**
This PR solves #2006
depends on https://github.com/tsuru/deploy-agent/pull/13

### CODE REVIEW

File: provision/kubernetes/builder_test.go [hunk 1]
```diff
@@ -280,3 +286,34 @@ cat >/dev/null && /bin/deploy-agent`)
...
+		c.Assert(containers[0].Env, check.DeepEquals, []apiv1.EnvVar{
+			{Name: "DEPLOYAGENT_RUN_AS_SIDECAR", Value: "true"},
+			{Name: "DEPLOYAGENT_DESTINATION_IMAGES", Value: "tsuru/myplatform:latest"},
+			{Name: "DEPLOYAGENT_SOURCE_IMAGE", Value: ""},
+			{Name: "DEPLOYAGENT_REGISTRY_AUTH_USER", Value: ""},
+			{Name: "DEPLOYAGENT_REGISTRY_AUTH_PASS", Value: ""},
+			{Name: "DEPLOYAGENT_REGISTRY_ADDRESS", Value: ""},
+			{Name: "DEPLOYAGENT_INPUT_FILE", Value: "/data/context.tar.gz"},
+			{Name: "DEPLOYAGENT_RUN_AS_USER", Value: "1000"},
+			{Name: "DEPLOYAGENT_IS_FILE_BUILD", Value: "true"},
```
Comment (andrestc) [inline provision/kubernetes/builder_test.go]: I think renaming this to `DEPLOYAGENT_DOCKERFILE_BUILD` would make this clearer

File: provision/kubernetes/deploy.go [hunk 2]
```diff
@@ -1136,6 +1234,7 @@ func (c deployAgentConfig) asEnvs() []apiv1.EnvVar {
 		{Name: "DEPLOYAGENT_REGISTRY_ADDRESS", Value: c.registryAddress},
 		{Name: "DEPLOYAGENT_INPUT_FILE", Value: c.inputFile},
 		{Name: "DEPLOYAGENT_RUN_AS_USER", Value: c.runAsUser},
+		{Name: "DEPLOYAGENT_IS_FILE_BUILD", Value: strconv.FormatBool(c.isFileBuild)},
```
Comment (andrestc) [inline provision/kubernetes/deploy.go]: I think renaming this to `DEPLOYAGENT_DOCKERFILE_BUILD` would make this clearer

File: builder/kubernetes/platform.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	return nil
+}
+
+func (b *kubernetesBuilder) buildPlatform(name string, args map[string]string, w io.Writer, data []byte, ctx context.Context) error {
+	client, err := getKubeClient()
+	if err != nil {
+		return err
+	}
+	var buf bytes.Buffer
+	writer := tar.NewWriter(&buf)
```
Comment (andrestc) [inline builder/kubernetes/platform.go]: I think we could extract the tar creation logic to a common place shared by the builders

File: builder/docker/platform.go [hunk 4]
```diff
@@ -28,52 +26,54 @@ import (
...
+
+func (b *dockerBuilder) buildPlatform(name string, args map[string]string, w io.Writer, data []byte) error {
 	client, err := getDockerClient()
 	if err != nil {
 		return err
 	}
+	var inputStream io.Reader
+	var buf bytes.Buffer
+	writer := tar.NewWriter(&buf)
+	writer.WriteHeader(&tar.Header{
```
Comment (andrestc) [inline builder/docker/platform.go]: I think we could extract the tar creation logic to a common place shared by the builders

**Comment by guilhermebr** [comment]
@andrestc PTAL

**Pull request merged**